API/Resource.md

```diff
@@ -0,0 +1,220 @@
+
+  /**
+   * Updates the server state with a PUT request
+   */
+  async put(options: PutRequestOptions<T> | State): Promise<void>; 
+
+  /**
+   * Deletes the resource
+   */
+  async delete(): Promise<void>; 
+
+  /**
+   * Sends a POST request to the resource.
+   *
+   * See the documentation for PostRequestOptions for more details.
+   * This function is used for RPC-like endpoints and form submissions.
+   *
+   * This function will return the response as a State object.
+   */
+  async post(options: PostRequestOptions): Promise<State>; 
+
+  /**
+   * Sends a POST request, and follows to the next resource.
+   *
+   * If a server responds with a 201 Status code and a Location header,
+   * it will automatically return the newly created resource.
+   *
+   * If the server responded with a 204 or 205, this function will return
+   * `this`.
+   */
+  async postFollow(options: PostRequestOptions): Promise<Resource>; 
+
+  /**
+   * Sends a PATCH request to the resource.
+   *
+   * This function defaults to a application/json content-type header.
+   */
+  async patch(options: PatchRequestOptions): Promise<void>; 
+
+  /**
+   * Follows a relationship, based on its reltype. For example, this might be
+   * 'alternate', 'item', 'edit' or a custom url-based one.
+   *
+   * This function can also follow templated uris. You can specify uri
+   * variables in the optional variables argument.
+   */
+  follow<TFollowedResource = any>(rel: string, variables?: LinkVariables): FollowPromiseOne<TFollowedResource>; 
+
+  /**
+   * Follows a relationship based on its reltype. This function returns a
+   * Promise that resolves to an array of Resource objects.
+   *
+   * If no resources were found, the array will be empty.
+   */
+  followAll<TFollowedResource = any>(rel: string): FollowPromiseMany<TFollowedResource>; 
+
+  /**
+   * Resolves a new resource based on a relative uri.
+   *
+   * Use this function to manually get a Resource object via a uri. The uri
+   * will be resolved based on the uri of the current resource.
+   *
+   * This function doesn't do any HTTP requests.
+   */
+  go<TGoResource = any>(uri: string): Resource<TGoResource>; 
+
+  /**
+   * Does a HTTP request on the current resource URI
+   */
+  fetch(init?: RequestInit): Promise<Response>; 
+
+  /**
+   * Does a HTTP request on the current resource URI.
+   *
+   * If the response was a 4XX or 5XX, this function will throw
+   * an exception.
+   */
+  fetchOrThrow(init?: RequestInit): Promise<Response>; 
+
+  /**
+   * Updates the state cache, and emits events.
+   *
+   * This will update the local state but *not* update the server
+   */
+  updateCache(state: State<T>); 
+
+  /**
+   * Clears the state cache for this resource.
+   */
+  clearCache(): void; 
+
+  /**
+   * Returns a Link object, by its REL.
+   *
+   * If the link does not exist, a LinkNotFound error will be thrown.
+   *
+   * @deprecated
+   */
+  async link(rel: string): Promise<Link>; 
```

Comment:
This is just a question, if someone who's been using Ketting 5.x and upgraded to 6.x, then used `link`, will this throw an error with the instruction what it has been changed to?

Reply:
No it will keep working. I think if you have a very up to date VS Code and typescript engine, I believe vscode might visually indicate that it's deprecated.